### ISSUE TITLE
Bug? - onCompletion triggering on function call, not on final response

### ISSUE THREAD

**Issue by RobertHH-IS** [vercel/ai#462]
I am using ai version 2.2.1
Here is simple code using functions and onCompletion.  

```
import { OpenAIStream, StreamingTextResponse } from 'ai'
import OpenAI from 'openai'
import { CompletionCreateParams } from 'openai/resources/chat'
// Create an OpenAI API client (that's edge friendly!)
const openai = new OpenAI({
  apiKey: process.env.OPENAI_API_KEY || ''
})

// IMPORTANT! Set the runtime to edge
export const runtime = 'edge'

const functions: CompletionCreateParams.Function[] = [
  {
    name: 'get_current_weather',
    description: 'Get the current weather',
    parameters: {
      type: 'object',
      properties: {
        format: {
          type: 'string',
          enum: ['celsius', 'fahrenheit'],
          description:
            'The temperature unit to use. Infer this from the users location.'
        }
      },
      required: ['format']
    }
  }
]

export async function POST(req: Request) {
  const { messages } = await req.json()
  const userMessage = messages[messages.length - 1].content
  console.log('userMessage', userMessage)

  const response = await openai.chat.completions.create({
    model: 'gpt-3.5-turbo-0613',
    stream: true,
    messages,
    functions
  })

  const stream = OpenAIStream(response, {
    experimental_onFunctionCall: async (
      { name, arguments: args },
      createFunctionCallMessages: any
    ) => {
      if (name === 'get_current_weather') {
        // Call a weather API here
        const weatherData = {
          temperature: 20,
          unit: args.format === 'celsius' ? 'C' : 'F'
        }
        const newMessages = createFunctionCallMessages(weatherData)
        console.log('newMessages', newMessages)
        return openai.chat.completions.create({
          messages: [...messages, ...newMessages],
          stream: true,
          model: 'gpt-3.5-turbo-0613',
          functions
        })
      }
    },
    onCompletion: async (completion: string) => {
      // This callback is called when the stream completes
      // You can use this to save the final completion to your database
      console.log('onCompletion called', completion)
    }
  })

  return new StreamingTextResponse(stream)
}
```

Here is the Chat.

![image](https://github.com/vercel/ai/assets/53995861/0e2f6917-03e3-4d9a-bc3f-e98f120bc2f4)

Here is the output of messages and onCompletion completion.  onCompletion takes the final output of normal chat calls, but only triggers on the first function call if a function is called.  The final response is not saved.  Expected behavior should be to trigger onCompletion for the final response.

userMessage Hi
onCompletion called Hello! How can I assist you today?
userMessage What is the weather today
onCompletion called {"function_call": {"name": "get_current_weather", "arguments": "{\n\"format\": \"celsius\"\n}"}}
newMessages [
  {
  role: 'assistant',
  content: '',
  function_call: { name: 'get_current_weather', arguments: '{\n"format": "celsius"\n}' }
},
  {
  role: 'function',
  name: 'get_current_weather',
  content: '{"temperature":20,"unit":"C"}'
}
]

**Comment by shuding** [comment]
#425 should fix this with a new API.

**Comment by MaxLeiter** [comment]
Fixed by #425.

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
This is a concrete callback behavior failure: the completion hook captures an intermediate function call rather than the final streamed response. Maintainer comments confirm it was fixed.

### Evidence
- With AI SDK 2.2.1, `onCompletion` fires for the intermediate function call payload instead of the final assistant response.
- The displayed chat receives the final response (“The temperature today is 20°C”), but that response is not passed to `onCompletion` for persistence.
- Maintainers state that PR #425 fixed the issue with a new API.
